src/levelset/levelSetSegmentationObject.cpp

```diff
@@ -3151,8 +3152,14 @@ void LevelSetSegmentationObject::_evalProjection(const std::array<double,3> &poi
 
     // If an unsigned evaluation is requested, the orientation of the surface should be discarded
     // and in order to have a normal that is agnostic with respect the two sides of the surface.
+    // The sign computation requires the projection reevaluation!
     if (!signedLevelSet) {
-        (*projectionNormal) *= static_cast<double>(evalSign(point));
+        short sign = evalSign(point);
+        if (sign == 0) {
+            throw std::runtime_error("Cannot evaluate the signed projection becuase the given point lays on the zero-level-set surface");
+        } else {
+            (*projectionNormal) *= static_cast<double>(sign);
+        }
```

Comment:
Don't throw an exception, otherwise the function will not be able to evaluate the projection of points on the zero-level-set surface, which can be performed without problems if the normal is not needed. Just do two changes:
 - allow to pass null pointers as arguments and perform the needed calculation only when the needed pointer are not null;
 - update the documentation of the function:
 ```
* \param[out] projectionPoint if a valid pointer is provided, out output it will contain the coordinates of the projection point on the surface
 * \param[out] projectionNormal if a valid pointer is provided, out output it will contain the normal to the surface on the projection point. If the projection point lies on the zero-level-set surface, and empty vector is returned
```

Reply:
Also, maybe find a better commit message.

Reply:
Done